src/posix/platform/udp.cpp

```diff
@@ -276,9 +276,6 @@ otError otPlatUdpBind(otUdpSocket *aUdpSocket)
         VerifyOrExit(0 == setsockopt(fd, IPPROTO_IPV6, IPV6_RECVPKTINFO, &on, sizeof(on)), error = OT_ERROR_FAILED);
     }
 
-    VerifyOrExit(0 == setsockopt(fd, IPPROTO_IPV6, IPV6_MULTICAST_IF, &gNetifIndex, sizeof(gNetifIndex)),
```

Comment:
Some explanation for removing this line of code:

- For sockets that bound to Thread network interface, this line of code is useless since the multicast network interface is already fixed to Thread interface only. 
- For sockets that bound to Backbone network interface, this line of code is wrong (will always fail) and thus should be removed. 
- For sockets that bound to unspecified (all) interfaces (e.g. mApplicationCoap), it could be a security pitfall to bind `MULTICAST_IF` to Thread interface because we might accidentally send multicast in the Thread network. I prefer to add `MUTLICAST_IF` configuration when we really need it.

Reply:
I'm fine get back to this issue later.